Sources/ClientRuntime/Networking/Http/URLSession/URLSessionHTTPClient.swift

```diff
@@ -53,8 +53,29 @@ public final class URLSessionHTTPClient: HTTPClient {
         /// The continuation for the asynchronous call that was made to initiate this request.
         ///
         /// Once the initial response is received, the continuation is called, and is subsequently set to `nil` so its
-        /// resources may be deallocated.
-        var continuation: CheckedContinuation<HttpResponse, Error>?
+        /// resources may be deallocated and to prevent it from being resumed twice.
+        private var continuation: CheckedContinuation<HttpResponse, Error>?
+
+        /// Returns `true` once the continuation is set to `nil`, which will happen once it has been resumed.
+        var hasBeenResumed: Bool { continuation == nil }
```

Comment:
Q: Is there a possibility for multiple threads accessing this value in a non thread-safe manner?

Reply:
After initial creation, connections are only accessed through methods on the `Storage` actor, which enforce exclusive access.